spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -115,6 +115,10 @@ public String wapId() {
     return sessionConf.get("spark.wap.id", null);
   }
 
+  public Integer outputSpecId() {
```

Comment:
Initially we were not sure whether to make spec IDs public but I also don't see a good alternative.
I am OK with the overall idea of exposing this.